clang/include/clang/Driver/Options.td

```diff
@@ -1041,28 +1041,15 @@ defm offload_uniform_block : BoolFOption<"offload-uniform-block",
   NegFlag<SetFalse, [], [ClangOption, CC1Option], "Don't assume">,
   BothFlags<[], [ClangOption], " that kernels are launched with uniform block sizes (default true for CUDA/HIP and false otherwise)">>;
 
-def fcx_limited_range : Joined<["-"], "fcx-limited-range">,
```

Comment:
The problem with aliasing is that the user would be allowed to write something like this:
-fcx-limited-range -fcomplex-arithmetic=improved
This will generate a warning like this:
warning: overriding '-fcomplex-arithmetic=basic' option with '-fcomplex-arithmetic=improved' [-Woverriding-option]

This warning is a bit mis-leading and doesn't reflect the option used in the command line. Not sure this can be corrected.

Reply:
Sorry. I meant "aliasing" in the non-technical sense of "having the same meaning." How that gets implemented is another matter. I think the driver could translate them to the same cc1 option.

Reply:
yes there is a way of doing that:
```
def fcx_limited_range : Flag<["-"], "fcx-limited-range">,
  Group<f_Group>, Visibility<[ClangOption, CC1Option]>,
  HelpText<"Basic algebraic expansions of complex arithmetic operations "
           "involving are enabled.">,
  Alias<fcomplex_arithmetic_EQ>, AliasArgs<["basic"]>;
```

That still produces the misleading warning for: -fcx-limited-range -fcomplex-arithmetic=improved

Reply:
What I meant to suggest is that you can leave the driver-level options as if they were independent, but when we process them in RenderFloatingPointOptions, -fcx-limited-range and -fcomplex-arithmetic=basic (for example), would add the same cc1 option. Since the warning is generated from the RenderFloatingPointOptions we should be able to make that report the expected output.